plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -258,17 +277,65 @@ public open class DefaultPageCreator(
                 headers = listOf(
                     headers("Name")
                 )
-            ) {
-                val documentations = it.sourceSets.map { platform ->
-                    it.descriptions[platform]?.also { it.root }
+            ) { pkg ->
+                val comment = pkg.sourceSets.mapNotNull { sourceSet ->
+                    pkg.descriptions[sourceSet]?.let { sourceSet to it }
+                }.distinctBy { it.second }.singleOrNull()
+
+                link(pkg.name, pkg.dri)
+                comment?.let { (sourceSet, description) ->
+                    createBriefComment(pkg, sourceSet, description)
                 }
-                val haveSameContent =
-                    documentations.all { it?.root == documentations.firstOrNull()?.root && it?.root != null }
+            }
 
-                link(it.name, it.dri)
-                if (it.sourceSets.size == 1 || (documentations.isNotEmpty() && haveSameContent)) {
-                    documentations.first()?.let { firstParagraphComment(kind = ContentKind.Comment, content = it.root) }
-                }
+            val hasTypes = m.packages.any {
+                it.classlikes.isNotEmpty() || it.typealiases.isNotEmpty()
+            }
+            if (hasTypes) {
+                header(2, "Index", kind = ContentKind.Cover)
+                link("All Types", AllTypesPageNode.DRI)
+            }
+        }
+    }
+
+    protected open fun contentForAllTypes(m: DModule): ContentGroup = contentBuilder.contentFor(m) {
+        group(kind = ContentKind.Cover) {
+            cover(m.name)
+        }
+
+        fun DTypelike.qualifiedName(): String {
+            val className = dri.classNames?.takeIf(String::isNotBlank) ?: name
+            val packageName = dri.packageName?.takeIf(String::isNotBlank) ?: return className
+            return "$packageName.${className}"
+        }
+
+        block(
+            name = "All Types",
+            level = 2,
+            kind = ContentKind.AllTypes,
+            elements = m.packages.flatMap { it.classlikes + it.typealiases }.filterOutActualTypeAlias(),
+            sourceSets = m.sourceSets.toSet(),
+            needsAnchors = true,
+            headers = listOf(
+                headers("Name")
+            )
+        ) { typelike ->
+            val comment = typelike.sourceSets.mapNotNull { sourceSet ->
+                typelike.descriptions[sourceSet]?.let { sourceSet to it }
+            }.distinctBy { it.second }.singleOrNull()
+
+            val customTags = typelike.customTags.values.mapNotNull { sourceSetTag ->
```

Comment:
As far as I see from code, author and throws are not custom tags.
In our codebase there are 2 custom tags provider: SinceKotlin and Mathjax
Still, may be we really need to show SinceKotlin only

Reply:
> As far as I see from code, author and throws are not custom tags.

Oh, sorry, I forgot(

>  Still, may be we really need to show SinceKotlin only

I think it should be consistent with a list of types on a regular page (package, classlike ..) so showing all custom tags is ok.

Reply:
@ilya-g what are your thoughts regarding comment above about best (more correct) way of showing `SinceKotlin`/description when they are different per source set?

Reply:
Thanks! Now we take minimum `SinceKotlin` version in All Types page. Implemented in https://github.com/Kotlin/dokka/pull/3267/commits/d744e49b1bf9fdcd25c4021492fa450c388cba22

Though, it's not affected by platform selector and will only show minimum version based on all platforms, as it looks like it's not possible to show different content based on different set of platform selectors.

Also I've checked how current stdlib documentation behaves, and looks like it doesn't show `SinceKotlin` at all if they are different per sourceSet + no changes after filtering by platform.
Here is an example for [kotlin.ConcurrentModificationException](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-concurrent-modification-exception/index.html):

<details>
<summary>With all platforms selected</summary>
<img width="738" alt="image" src="https://github.com/Kotlin/dokka/assets/50462752/614fd1be-41eb-45cb-9248-30af1aad0228">
</details>

<details>
<summary>With only Native selected</summary>
<img width="752" alt="image" src="https://github.com/Kotlin/dokka/assets/50462752/9b1c1d33-520c-49db-8ed9-2d6d5413061e">
</details>

So I would propose to go for now **without** additional filtering based on selected platforms. And implement it later if needed. @ilya-g WDYT?

Reply:
We discussed this question with Oleg, and it looks like we indeed don't have the technical ability to change the value dynamically based on platform selectors.

Implementing it would take considerable time, and it doesn't seem to be that critical for the initial release, so let's just merge it with the currently implemented and consistent logic: always show the min version

We can change it later on if it becomes apparent that it's misleading or is a problem